x/wasm/client/cli/tx_test.go

```diff
@@ -57,3 +63,53 @@ func TestParseAccessConfigFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestParseStoreCodeFlags(t *testing.T) {
+	anyValidWasmFile, err := os.CreateTemp(t.TempDir(), "wasm")
+	require.NoError(t, err)
+
+	anyValidWasmFile.Write(wasmIdent)
+
+	wasmBin, err := io.ReadAll(anyValidWasmFile)
+	require.NoError(t, err)
+
+	checkSum := sha256.Sum256(wasmBin)
+	checksumStr := hex.EncodeToString(checkSum[:])
+
+	require.NoError(t, anyValidWasmFile.Close())
+	specs := map[string]struct {
+		args   []string
+		expCfg *types.AccessConfig
+		expErr bool
+	}{
+		"source missing": {
+			args:   []string{"--builder=foo", "--code-hash=" + checksumStr},
+			expErr: true,
+		},
+		"builder missing": {
+			args:   []string{"--source=bar", "--code-hash=" + checksumStr},
+			expErr: true,
+		},
+		"code hash missing": {
+			args:   []string{"--source=bar", "--builder=foo"},
+			expErr: true,
+		},
+		"happy path": {
+			args:   []string{"--source=bar", "--builder=foo", "--code-hash=" + checksumStr},
+			expErr: false,
+		},
```

Comment:
Please add a case where the source file is zipped so that you are not running into a problem when comparing the hash